src/Microsoft.AspNet.OData.Shared/ODataAPIHandlerFactory.cs

```diff
@@ -0,0 +1,53 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataAPIHandlerFactory.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Factory class for ODataAPIHandlers.
+    /// </summary>
+    public abstract class ODataAPIHandlerFactory
+    {
+        /// <summary>
+        /// Creates an instance of an ODataAPIHandlerFactory with the given model.
+        /// </summary>
+        /// <param name="model">The IEdmModel for the API Handler Factory.</param>
+        protected ODataAPIHandlerFactory(IEdmModel model)
+        {
+            Model = model;
+        }
+
+        /// <summary>
+        /// The IEdmModel for the Factory.
+        /// </summary>
+        public IEdmModel Model { get; private set; }
+
+        /// <summary>
+        /// Get the handler depending on OData path.
+        /// </summary>
+        /// <param name="odataPath">OData path corresponding to an odataid.</param>
```

Comment:
This being public documentation for this method, would most people be able to relate with "odataid"?